peachjam/adapters/indigo.py

```diff
@@ -510,6 +519,71 @@ def download_source_file(self, url, doc, title):
                 },
             )
 
+    def create_publication_file(self, publication_document, doc, title, stub=False):
+        from peachjam.models import PublicationFile
+
+        logger.info(f"Creating / updating a publication file for {title}")
+
+        # first delete any existing PublicationFile file: a new one will be saved if needed
+        # TODO: only delete it if something has changed? (how to know that a file has changed?)
+        if hasattr(doc, "publication_file"):
+            if doc.publication_file.file:
+                logger.info(
+                    f"  Deleting existing PublicationFile file on {doc.work_frbr_uri}"
+                )
+                doc.publication_file.file.delete()
+
+        if stub:
+            if hasattr(doc, "source_file"):
+                logger.info("  Stub: Using the source file")
+                PublicationFile.objects.update_or_create(
+                    document=doc,
+                    defaults={
```

Comment:
I think you also want to set `url` to None, otherwise it could have an old value (even though it is not used).

Reply:
oh, `clean()` will sort it out.